File: 0027-remove-element/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        
        vector<int>res;
        // int idx = 0;
        for(int i = 0; i < nums.size(); i++){
            if(nums[i] != val){
                res.push_back(nums[i]);
                
            }
        }

        nums = res;
        return nums.size();
    }
};

File: 0027-remove-element/0027-remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        
        int count = 0 , ans = 0;
        for(int i = 0 ; i < nums.length ; i++){
            if(nums[i] == val){
                count++;
            }else{
                ans++;
                int temp = nums[i];
                nums[i] = nums[i - count];
                nums[i - count] = temp;
            }
        }
        return ans;
    }
}